osc/commandline.py

```diff
@@ -6404,21 +6404,27 @@ def parse_repoarchdescr(self, args, noinit = False, alternative_project = None,
                 # only persist our own repos
                 Repo.tofile(repolistfile, repositories)
 
-        no_repo = False
-        repo_names = sorted(set([r.name for r in repositories]))
-        if not arg_repository and repositories:
-            # XXX: we should avoid hardcoding repository names
-            # Use a default value from config, but just even if it's available
-            # unless try standard, or openSUSE_Factory, or openSUSE_Tumbleweed
-            no_repo = True
-            arg_repository = repositories[-1].name
-            for repository in (conf.config['build_repository'], 'standard', 'openSUSE_Factory', 'openSUSE_Tumbleweed'):
-                if repository in repo_names:
-                    arg_repository = repository
-                    no_repo = False
-                    break
+        no_repo = not not arg_repository
```

Comment:
This double negation seems to be an error.
Shouldn't this rather be something like `no_repo = arg_repository is None`?
There are also 2 "unreachable statement" errors[1] reported by LGTM, but they can be related to this line.
[1] https://lgtm.com/projects/g/openSUSE/osc/rev/pr-e5a8971fc371d06403bf59466faebc2add25846a

Reply:
Double negation is a perfectly fine way to turn something into a boolean.

I am not even sure that `is None` has the same semantic.

The unreachable statements are at least in part due to the no_repo usage cleanup.